karapace/protobuf/schema.py

```diff
@@ -1,14 +1,188 @@
+from karapace.protobuf.utils import append_documentation, append_indented
 
+import logging
 
+log = logging.getLogger(__name__)
+
+
+def static_init(cls):
+    if getattr(cls, "static_init", None):
+        cls.static_init()
+    return cls
+
+
+def add_slashes(text: str):
+    escape_dict = {
+        r'\a': '\\a',
+        r'\b': '\\b',
+        r'\c': '\\c',
+        r'\f': '\\f',
+        r'\n': '\\n',
+        r'\r': '\\r',
+        r'\t': '\\t',
+        r'\v': '\\v',
+        '\'': "\\'",
+        '\"': '\\\"',
+        '\\': '\\\\'
+    }
+    result: str = ""
+    for char in text:
+        try:
+            result += escape_dict[char]
+        except KeyError:
+            result += char
+    return result
+
+
+def message_element_string(element: MessageElement) -> str:
+    result: list = list()
+    append_documentation(result, element.documentation)
+    result.append(f"message {element.name} {{")
+    if element.reserveds:
+        result.append("\n")
+        for reserved in element.reserveds:
+            append_indented(result, reserved.to_schema())
+
+    if element.options:
+        result.append("\n")
+        for option in element.options:
+            append_indented(result, option_element_string(option))
+
+    if element.fields:
+        result.append("\n")
+        for field in element.fields:
+            append_indented(result, field.to_schema())
+
+    if element.one_ofs:
+        result.append("\n")
+        for one_of in element.one_ofs:
+            append_indented(result, one_of.to_schema())
+
+    if element.groups:
+        result.append("\n")
+        for group in element.groups:
+            append_indented(result, group.to_schema())
+
+    if element.extensions:
+        result.append("\n")
+        for extension in element.extensions:
+            append_indented(result, extension.to_schema())
+
+    if element.nested_types:
+        result.append("\n")
+        for nested_type in element.nested_types:
+            if isinstance(nested_type, MessageElement):
+                append_indented(result, message_element_string(nested_type))
+
+        for nested_type in element.nested_types:
+            if isinstance(nested_type, EnumElement):
+                append_indented(result, enum_element_string(nested_type))
+
+    result.append("}\n")
+    return "".join(result)
+
+
+def enum_element_string(element: EnumElement) -> str:
+    return element.to_schema()
+
+
+def option_element_string(option: OptionElement):
+    result: str
+    if option.kind == OptionElement.Kind.STRING:
+        name: str
+        if option.is_parenthesized:
+            name = f"({option.name})"
+        else:
+            name = option.name
+        value = add_slashes(str(option.value))
+        result = f"{name} = \"{value}\""
+    else:
+        result = option.to_schema()
+
+    return f"option {result};\n"
+
+
+@static_init
 class ProtobufSchema:
+    @classmethod
+    def static_init(cls):
```

Comment:
It seems that `static_init` decorator and methods are not really necessary. From what I gathered it should be sufficient to do:

```python
class ProtobufSchema:
    DEFAULT_LOCATION = Location.get("")